src/main/java/com/icthh/xm/ms/otp/config/SecurityConfiguration.java

```diff
@@ -44,6 +48,9 @@ public void configure(HttpSecurity http) throws Exception {
             .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
         .and()
             .authorizeRequests()
+            .antMatchers("/api/login").permitAll()
+            .antMatchers("/api/userinfo").permitAll()
+            .antMatchers("/api/oauth/token").permitAll()
```

Comment:
Why do we permiAll for this endpoint?

Reply:
i assume we cannot avoid permitting api/login coz user does not have any permission/token before interacting with our system. If i remove next 2 endpoints, add 2 @PreAuthorize annotations and give necessary permissions on web UI i get 401 when uaa communicating with otp. Looks like these requests require some additional token(s). From the other hand am not sure that these endpoints  should be protected: first just convert `code` parameter into `access_token` and second  retrieves user data from jwt token and sends it back to uaa.